aws/resource_aws_backup_plan.go

```diff
@@ -248,66 +175,64 @@ func resourceAwsBackupPlanUpdate(d *schema.ResourceData, meta interface{}) error
 func resourceAwsBackupPlanDelete(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).backupconn
 
-	input := &backup.DeleteBackupPlanInput{
+	log.Printf("[DEBUG] Deleting Backup Plan: %s", d.Id())
+	_, err := conn.DeleteBackupPlan(&backup.DeleteBackupPlanInput{
 		BackupPlanId: aws.String(d.Id()),
-	}
-
-	_, err := conn.DeleteBackupPlan(input)
+	})
 	if isAWSErr(err, backup.ErrCodeResourceNotFoundException, "") {
 		return nil
 	}
 
 	if err != nil {
-		return fmt.Errorf("error deleting Backup Plan: %s", err)
+		return fmt.Errorf("error deleting Backup Plan (%s): %s", d.Id(), err)
 	}
 
 	return nil
 }
 
-func expandBackupPlanRules(l []interface{}) []*backup.RuleInput {
+func expandBackupPlanRules(vRules *schema.Set) []*backup.RuleInput {
 	rules := []*backup.RuleInput{}
 
-	for _, i := range l {
-		item := i.(map[string]interface{})
+	for _, vRule := range vRules.List() {
 		rule := &backup.RuleInput{}
 
-		if item["rule_name"] != "" {
-			rule.RuleName = aws.String(item["rule_name"].(string))
+		mRule := vRule.(map[string]interface{})
+
+		if vRuleName, ok := mRule["rule_name"].(string); ok && vRuleName != "" {
+			rule.RuleName = aws.String(vRuleName)
+		} else {
+			continue
```

Comment:
This is the same scenario as commented on here: https://github.com/terraform-providers/terraform-provider-aws/pull/6095/files#r226842239.